api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -632,7 +632,6 @@ type HostedClusterSpec struct {
 	// TODO: key/value validations break cost budget for <=4.17. We should figure why and enable it back.
 	// +kubebuilder:validation:MaxProperties=20
 	// +optional
-	// +openshift:enable:FeatureGate=HCPPodsLabels
```

Comment:
can we include a test here https://github.com/openshift/hypershift/blob/main/test/e2e/create_cluster_test.go#L47 that verifies this validation?

Reply:
done